src/main/java/com/qouteall/immersive_portals/portal/Portal.java

```diff
@@ -624,4 +628,19 @@ public Vec3d getPointInPortalProjection(Vec3d pos) {
             axisH.multiply(yInPlane)
         );
     }
+
+    /**
+     * @return The {@link World} of this portal's {@link #dimensionTo}.
+     */
+    public World getDestinationWorld(EnvType env) {
```

Comment:
??? It should use world.isClient because the internal server and client both exist in singleplayer

Reply:
And all code using Fabric specific things should be put into hiding_in_the_bushes

Reply:
It does use world.isClient, that is, rayTrace passes an EnvType based on that

Reply:
Changed it to accept boolean instead of EnvType